ci/sync_envoy.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash -e
+
+set -o pipefail
+
+MIRROR_MSG="Mirrored from envoyproxy/envoy"
+SRCS=(envoy contrib)
+GO_TARGETS=(@envoy_api//...)
+IMPORT_BASE="github.com/envoyproxy/go-control-plane"
+COMMITTER_NAME="envoy-sync[bot]"
+COMMITTER_EMAIL="envoy-sync[bot]@users.noreply.github.com"
+
+
+if [[ -z "$ENVOY_SRC_DIR" ]]; then
+    echo "ENVOY_SRC_DIR not set, it should point to a cloned Envoy repo" >&2
+    exit 1
+elif [[ ! -e "$ENVOY_SRC_DIR" ]]; then
+    echo "ENVOY_SRC_DIR ($ENVOY_SRC_DIR) not found, did you clone it?" >&2
+    exit 1
+fi
+
+
+build_protos () {
+    # TODO(phlax): use envoy do_ci target once https://github.com/envoyproxy/envoy/pull/27675 lands
+    local go_protos go_proto go_file rule_dir proto input_dir output_dir
+    echo "Building go protos ..."
+    cd "${ENVOY_SRC_DIR}" || exit 1
+
+    # shellcheck disable=SC1091
+    . ci/setup_cache.sh
+
+    read -a go_protos <<< "$(bazel query "kind('go_proto_library', ${GO_TARGETS[*]})" | tr '\n' ' ')"
```

Comment:
nit:
```
$ shellcheck ci/sync_envoy.sh

In ci/sync_envoy.sh line 31:
    read -a go_protos <<< "$(bazel query "kind('go_proto_library', ${GO_TARGETS[*]})" | tr '\n' ' ')"
    ^--^ SC2162 (info): read without -r will mangle backslashes.

For more information:
  https://www.shellcheck.net/wiki/SC2162 -- read without -r will mangle backs...
```

Reply:
would be good to add shellcheck to the ci here - i was running it locally before submitting but i think this slipped through

also, would be good to clear up the whitespace issues 

```console
$ pip install envoy.code.check
$ envoy.code.check . -c glint
```

will show you issues